_test/test_sqw/test_PixelData_operations.m

```diff
@@ -71,17 +75,13 @@ function delete(obj)
     end
 
     function setUp(obj)
-        obj.config = hor_config;
-        obj.old_config = obj.config.get_data_to_store();
-    end
-
-    function tearDown(obj)
-        set(hor_config, obj.old_config);
+        obj.pix_in_memory = copy(obj.pix_in_memory_base);
+        obj.pix_with_pages = copy(obj.pix_with_pages_base);
     end
 
     function test_compute_bin_data_correct_output_in_memory_mex_1_thread(obj)
-        obj.config.use_mex = true;
-        obj.config.threads = 1;
+        cleanup = set_config_option(hor_config(), {'use_mex', 'threads'}, ...
+                                    {true, 1});
 
```

Comment:
It appears that cleanup is only set because set_config_option forces this. Wondering if this restriction should be removed for set_config_option. Alternatively the output could be renamed dummy or not_used or some convention. Even a comment to make this clear could prevent code searching to find out what is happening.

Reply:
Further inspection - action of cleanup appears to take effect on going out of scope. Is there a way of making this clearer?

Reply:
You're right, the intention of the function could be clearer. It was intended that `set_config_option` be used purely as a test helper (it lives in `_test`, so is only available to be used by tests). It sets some config values and returns an [`onCleanup`](https://www.mathworks.com/help/matlab/ref/oncleanup.html) object which, when it goes out of scope, will return the config to its original state. 

I'm struggling to come up with a good name. `set_config_option_with_cleanup`?

Reply:
`cleanup_handler = set_temporary_config_options( ... )`?